tests/components/osoenergy/test_water_heater.py

```diff
@@ -0,0 +1,260 @@
+"""The water heater tests for the OSO Energy platform."""
+
+from unittest.mock import ANY, MagicMock, patch
+
+from syrupy import SnapshotAssertion
+
+from homeassistant.components.osoenergy.const import DOMAIN
+from homeassistant.components.osoenergy.water_heater import (
+    ATTR_UNTIL_TEMP_LIMIT,
+    ATTR_V40MIN,
+    SERVICE_GET_PROFILE,
+    SERVICE_SET_PROFILE,
+    SERVICE_SET_V40MIN,
+)
+from homeassistant.components.water_heater import (
+    DOMAIN as WATER_HEATER_DOMAIN,
+    SERVICE_SET_TEMPERATURE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_TEMPERATURE,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from tests.common import snapshot_platform
+
+
+@patch("homeassistant.components.osoenergy.PLATFORMS", [Platform.WATER_HEATER])
+async def test_water_heater(
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+    mock_osoenergy_client: MagicMock,
+    snapshot: SnapshotAssertion,
+    mock_config_entry: ConfigEntry,
+) -> None:
+    """Test states of the water heater."""
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
+
+
+async def test_get_profile(
+    hass: HomeAssistant,
+    mock_osoenergy_client: MagicMock,
+    mock_config_entry: ConfigEntry,
+) -> None:
+    """Test getting the heater profile."""
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    profile = await hass.services.async_call(
+        DOMAIN,
+        SERVICE_GET_PROFILE,
+        {ATTR_ENTITY_ID: "water_heater.test_device"},
+        blocking=True,
+        return_response=True,
+    )
+    assert profile.get("water_heater.test_device").get("profile") == [
+        60,
+        60,
+        60,
+        60,
+        60,
+        60,
+        60,
+        60,
+        60,
+        60,
+        60,
+        60,
+        60,
+        60,
+        60,
+        60,
+        60,
+        10,
+        60,
+        60,
+        60,
+        60,
+        60,
+        60,
+    ]
```

Comment:
let's assert the full entity response:
```suggestion
    assert profile == {
        "water_heater.test_device": {
            "profile": [
                60,
                60,
                60,
                60,
                60,
                60,
                60,
                60,
                60,
                60,
                60,
                60,
                60,
                60,
                60,
                60,
                60,
                10,
                60,
                60,
                60,
                60,
                60,
                60,
            ],
        },
    }
```